server/mailrepository/mailrepository-api/src/main/java/org/apache/james/mailrepository/api/MailRepository.java

```diff
@@ -59,6 +163,18 @@ default Publisher<Long> sizeReactive() {
      */
     Iterator<MailKey> list() throws MessagingException;
 
+    default Iterator<MailKey> list(List<Condition> conditions) throws MessagingException {
```

Comment:
We would be more flexible by creating a `And` condition and passing a single condition here?